packages/client/graph-scaffolder/examples/src/index.ts

```diff
@@ -31,8 +31,18 @@ class SampleRenderer extends BasicRenderer<NodeData, EdgeData> {
 	}
 
 	renderNodes(selection: D3SelectionINode<NodeData>) {
-		const species = selection.filter((d) => d.data.type === 'species');
-		const transitions = selection.filter((d) => d.data.type === 'transition');
+		const species = selection.filter((d) => d.type !== 'custom' && d.data.type === 'species');
+		const transitions = selection.filter(
+			(d) => d.type !== 'custom' && d.data.type === 'transition'
+		);
+		const customs = selection.filter((d) => d.type === 'custom');
```

Comment:
I can't wait until we can use [`Array.prototype.group()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/group)
```js
const { species, transitions, customs } = selection.group(({ type }) => type);
```

Until then why not use [`Array.prototype.reduce()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce#grouping_objects_by_a_property)
```js
const { species, transitions, customs } = selection.reduce((sorted, d) => {
    const curGroup = sorted[d.type] ?? [];
    return { ...sorted, [d.type]: [...curGroup, d] };
}, {});
```

Reply:
Because I do not like the reduce -syntax, I find them harder to read, and there is not sufficient data here where a reduce call will have a faster performance than filter 🤷